packages/api/lambdas/write-db-dlq-records-to-s3.js

```diff
@@ -1,43 +1,142 @@
+//@ts-check
+
 'use strict';
 
 const get = require('lodash/get');
 const uuidv4 = require('uuid/v4');
 
 const log = require('@cumulus/common/log');
+const { isEventBridgeEvent } = require('@cumulus/aws-client/Lambda');
 const { s3PutObject } = require('@cumulus/aws-client/S3');
 const { parseSQSMessageBody } = require('@cumulus/aws-client/SQS');
-const { getMessageExecutionName } = require('@cumulus/message/Executions');
-const { unwrapDeadLetterCumulusMessage } = require('@cumulus/message/DeadLetterMessage');
-
-function determineExecutionName(cumulusMessageObject) {
-  try {
-    return getMessageExecutionName(cumulusMessageObject);
-  } catch (error) {
-    log.error('Could not find execution name in cumulus_meta:', cumulusMessageObject.cumulus_meta);
-    return 'unknown';
+const { unwrapDeadLetterCumulusMessage, isSQSRecordLike } = require('@cumulus/message/DeadLetterMessage');
+const { getCumulusMessageFromExecutionEvent } = require('@cumulus/message/StepFunctions');
+/**
+ *
+ * @typedef {import('@cumulus/types/message').CumulusMessage} CumulusMessage
+ * @typedef {import('@cumulus/types').MessageGranule} MessageGranule
+ * @typedef {{granules: Array<MessageGranule>}} PayloadWithGranules
+ * @typedef {import('@cumulus/message/DeadLetterMessage').DLQRecord} DLQRecord
+ * @typedef {import('aws-lambda').EventBridgeEvent} EventBridgeEvent
+ */
+
+/**
+ * @param {unknown} payload
+ * @returns {payload is PayloadWithGranules}
+ */
+function payloadHasGranules(payload) {
+  return (
+    payload instanceof Object
+    && 'granules' in payload
+    && Array.isArray(payload.granules)
+  );
+}
+/**
+ * Reformat object with key attributes at top level.
+ *
+ * @param {DLQRecord | EventBridgeEvent} dlqRecord - event bridge event as defined in aws-lambda
+ * @returns {Promise<Object>} - message packaged with
+ * metadata or null where metadata not found
+ * {
+ *   error: <errorString | null>
+ *   time: <timestamp(utc) | null>
+ *   status: <status | null>
+ *   collection: <collectionName | null>
+ *   granules: <[granuleIds, ...] | []>
+ *   execution: <executionArn | null>
+ *   stateMachine: <stateMachineArn | null>
+ *   ...originalAttributes
+ * }
+ */
+async function hoistCumulusMessageDetails(dlqRecord) {
```

Comment:
This method looks like it's private-duplicating some of the internal logic of `unwrapDeadLetterCumulusMessage` and/or making some sensible refactors - is there a reason we can't/shouldn't reuse that code and leave it common in the message package,  particularly given the other single use of that call? 

If not, does this have adaquate unit coverage?

Reply:
Withdraw question re: units now that I've gotten to that file in the review :thumbsup:, the rest stands.

Reply:
(I think) while this has some structural similarity to unwrap DeadLetterMessage (it does type checking from within a subset of the same types) it is not the same. i'm looking into how they could be joined, but fundamentally:
1 of these extracts a cumulusMessage object from within a questionably structured object
the other extracts some specific data from within a questionably structured object and packages it with the o riginal object

the obvious join would be to refactor this to a function that extracts all of these things and lets you do what you wish with it such that I can either capture that cumulusMessage and do something with it or capture those other pieces of data and do something with them, but prototyping it a little bit that adds a lot of logical complexity while reducing the amount of code by not very much.
at least by the route I've come up with so far

Reply:
discussed with Jonathan and does appear to add too much complexity trying to make this refactor